test/e2e/framework/test_context.go

```diff
@@ -264,7 +266,7 @@ func RegisterClusterFlags() {
 	flag.StringVar(&TestContext.KubeVolumeDir, "volume-dir", "/var/lib/kubelet", "Path to the directory containing the kubelet volumes.")
 	flag.StringVar(&TestContext.CertDir, "cert-dir", "", "Path to the directory containing the certs. Default is empty, which doesn't use certs.")
 	flag.StringVar(&TestContext.RepoRoot, "repo-root", "../../", "Root directory of kubernetes repository, for finding test files.")
-	flag.StringVar(&TestContext.Provider, "provider", "", "The name of the Kubernetes provider (gce, gke, local, etc.)")
+	flag.StringVar(&TestContext.Provider, "provider", "", "The name of the Kubernetes provider (gce, gke, local, skeleton, etc.)")
```

Comment:
Why not default to skeleton here?

Reply:
so both `local` and `skeleton` are null providers here:
https://github.com/kubernetes/kubernetes/blob/b4384e9f828b97c380b93483bea279d2a71d1d00/test/e2e/framework/provider.go#L60-L66

if the previous fallback from `""` as provider was `local` then today setting to `skeleton` as default should not have a negative effect (famous last words).

`local` and `skeleton` in test-infra/kubetest on the other hand *are* different.

Reply:
The question is whether anyone cares about having this log message when `--provider` is not set:
https://github.com/kubernetes/kubernetes/blob/b4384e9f828b97c380b93483bea279d2a71d1d00/test/e2e/framework/provider.go#L69

Reply:
If we hard fail we should have a sane default imo.

Reply:
local is different here too, because of log dumping in the framework. skeleton is a sane default.

Reply:
so passing `local` here also does extra steps - e.g. logging? i didn't know that...
my thinking was that it only affects logs on the kubetest side.

Reply:
e2e.test will invoke the cluster logs dump script itself, which is """provider""" string aware, including local.

https://github.com/kubernetes/kubernetes/blob/5e86fa43f59f8ed4fd353ab17dc1e65708a9ae71/test/e2e/e2e.go#L154-L162

https://github.com/kubernetes/kubernetes/blob/5e86fa43f59f8ed4fd353ab17dc1e65708a9ae71/test/e2e/framework/util.go#L4760-L4768

Reply:
i'm convinced that `log-dump.sh` is evil at this point.

Reply:
@pohly so `""` already does not fallback to "local" if i'm not wrong, as per:
https://github.com/kubernetes/kubernetes/blob/b4384e9f828b97c380b93483bea279d2a71d1d00/test/e2e/framework/provider.go#L68-L71

it effectively falls back to a unnamed null-provider in 1.13 and master, so from what i'm seeing we already modified the old behavior.

@timothysc 
> If we hard fail we should have a sane default imo.

from my understanding we are not failing but instead we are only showing a warning if `--provider` is not set (equals `""`).

Reply:
@neolit123 right, it currently basically falls back to "`skeleton` plus warning". Forget what I said about falling back to "local". I think that the current behavior is consistent with <1.13 but haven't checked.

I'm in favor of removing this odd log message and just making `skeleton` the explicit default. Any objections?

Reply:
+1

Reply:
Done. See latest commit in this PR.